Mathlib/LinearAlgebra/Matrix/PosDef.lean

```diff
@@ -263,11 +263,10 @@ lemma posSemidef_iff_eq_transpose_mul_self {A : Matrix n n 𝕜} :
   rw [hA.posSemidef_sqrt.1]
 
 lemma IsHermitian.posSemidef_of_eigenvalues_nonneg [DecidableEq n] {A : Matrix n n 𝕜}
-    (hA : IsHermitian A) (h : ∀ i : n, 0 ≤ hA.eigenvalues i) : PosSemidef A := by
-  simp_rw [hA.conjTranspose_eigenvectorMatrix.symm ▸ hA.spectral_theorem']
-  refine (posSemidef_diagonal_iff.mpr fun i ↦ ?_).mul_mul_conjTranspose_same _
-  rw [RCLike.le_iff_re_im]
-  simpa using h i
+    (hA : IsHermitian A) (h : ∀ i : n, 0 ≤ hA.eigenvalues i) : PosSemidef A :=
+  hA.spectral_theorem2 ▸
+    (posSemidef_diagonal_iff.mpr
+    (by simpa [RCLike.ofReal_nonneg] using h)).mul_mul_conjTranspose_same _
```

Comment:
The bad line break positions suggest that we should ungolf this slightly.

Reply:
Replaced with your slightly ungolfed version of it.